src/libs/ShareManager/index.native.js

```diff
@@ -0,0 +1,82 @@
+import ShareMenu from 'react-native-share-menu';
+import {redirect} from '../actions/App';
+import {set as setSharedItem} from '../actions/SharedItem';
+import ROUTES from '../../ROUTES';
+import ShareType from './ShareType';
+import sharedItemPropTypes from './PropTypes';
+
+const TEXT_MIME_TYPE = /^text\//i;
+
+// Share event listener
+let listener;
+
+/**
+ * Prepare shared item for posting
+ *
+ * @param {Object} sharedItem Shared item object
+ * @returns {Object} Prepared shared item structure
+ */
+function prepareSharedItem(sharedItem) {
+    if (sharedItem.mimeType.match(TEXT_MIME_TYPE)) {
```

Comment:
If something doesn't match a `TEXT_MIME_TYPE` I think we should return early rather than fall to the `return` statement below

Reply:
Could you please elaborate?

Currently the code does following:
```
  if it's TEXT_MIME_TYPE
     then return TEXT object

  return FILE object
```

You mean it's better to return early, if it's not text?
```
  if it's not TEXT_MIME_TYPE
     then return FILE object

  return TEXT object
```